File: Patterns/Factory Method Pattern/Cpp/PizzaStore/NYPizzaStore.cpp
#include "../Pizza/Pizza.cpp"
#include "../Pizza/NYStylePizza/NYStyleCheesePizza.cpp"
#include "../Pizza/NYStylePizza/NYStyleClamPizza.cpp"
#include "../Pizza/NYStylePizza/NYStylePepperoniPizza.cpp"
#include "../Pizza/NYStylePizza/NYStyleVeggiePizza.cpp"

#include "PizzaStore.cpp"

class NYPizzaStore : public PizzaStore {
    Pizza* createPizza(string type) override {
        if (type == "cheese") return new NYStyleCheesePizza();
        else if (type == "clam") return new NYStyleClamPizza();
        else if (type == "pepperoni") return new NYStylePepperoniPizza();
        else if (type == "veggie") return new NYStyleVeggiePizza();
        else return nullptr;
    }
};

File: Patterns/Factory Method Pattern/CSharp/PizzaStore/NYPizzaStore.cs
using Pizza;

using NYStylePizza = Pizza.NYStylePizza;

namespace PizzaStore
{
    public class NYPizzaStore : PizzaStore 
    {
        protected override Pizza.Pizza? CreatePizza(string type)
        {
            switch (type) 
            {
                case "cheese" : return new NYStylePizza.NYStyleCheesePizza();
                case "clam" : return new NYStylePizza.NYStyleClamPizza();
                case "pepperoni" : return new NYStylePizza.NYStylePepperoniPizza();
                case "veggie" : return new NYStylePizza.NYStyleVeggiePizza();
            }

            return null;
        }
    }
}